File: shared/src/main/java/ru/privatenull/pnauth/limbo/LimboServer.java
package ru.privatenull.pnauth.limbo;

import java.util.Optional;

public interface LimboServer extends AutoCloseable {
    String id();

    String host();

    int port();

    LimboServerState state();

    default Optional<LimboControl> control() {
        return Optional.empty();
    }

    void start() throws Exception;

    void stop();

    @Override
    default void close() {
        stop();
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/limbo/LimboServer.kt
package ru.privatenull.pnauth.limbo

interface LimboServer : AutoCloseable {
    fun id(): String

    fun host(): String

    fun port(): Int

    fun state(): LimboServerState

    fun start()

    fun stop()

    override fun close() {
        stop()
    }
}
